src/main.cpp

```diff
@@ -1074,6 +1151,14 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         }
     }
 
+    int nextBlockHeight = chainActive.Height() + 1;
+
+    // FIXME: DoS level set to 10 to be more forgiving... or use 100?
+    // Check transaction contextually against the set of consensus rules which apply in the next block to be mined.
+    if (!ContextualCheckTransaction(tx, state, nextBlockHeight, 10)) {
```

Comment:
Do the contextual checks after the non-contextual ones (i.e. put this in between `CheckTransaction()` and `IsCoinBase()`).

Reply:
Updated mempool test for this and added new unit tests.